File: SpringBoot/src/main/java/com/hospital/qlbv/entity/Payment.java
package com.hospital.qlbv.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "payments")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Payment {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer paymentId;
    
    @Column(name = "patient_id")
    private Integer patientId;
    
    @Column(name = "appointment_id")
    private Integer appointmentId;
    
    private Double amount;
    
    @Column(name = "payment_method")
    private String paymentMethod;
    
    private String status;
    
    @Column(name = "transaction_id")
    private String transactionId;
    
    @Column(name = "payment_date")
    private String paymentDate;
    
    private String description;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "patient_id", insertable = false, updatable = false)
    @JsonIgnoreProperties({"user", "insurance", "appointments"})
    private Patient patient;
}

File: Backend/Models/Payment.cs
using System.ComponentModel.DataAnnotations;

namespace QuanLyBenhVien.API.Models;

public class Payment
{
    [Key]
    public int PaymentID { get; set; }
    public int? PatientID { get; set; }
    public int? AppointmentID { get; set; }
    public decimal? Amount { get; set; }
    public string? PaymentMethod { get; set; }
    public string? Status { get; set; }
    public string? TransactionID { get; set; }
    public DateTime? PaymentDate { get; set; }
    public string? Description { get; set; }

    // Navigation properties
    public virtual Patient? Patient { get; set; }
    public virtual Appointment? Appointment { get; set; }
}
